### ISSUE TITLE
Joining efforts

### ISSUE THREAD

**Issue by leojrfs** [jpommerening/homebrew-avr#8]
There are plenty repos as:
https://github.com/jpommerening/homebrew-avr
https://github.com/larsimmisch/homebrew-avr
https://github.com/WeAreLeka/homebrew-avr

What do you think of joining efforts and avoiding duplication, by creating an org like homebrew-uC/homebrew-avr

And then we can also join suff from other uC archs :smiley: 

i am also posting the same issue on the listed repos

**Comment by jpommerening** [comment]
That's a great idea! I would love to have a couple people reviewing my patches!

We'll need a process to make sure we don't step on each others toes. I don't want to be commiting something that breaks the build for everyone else :sweat_smile:

I'd propose something along the lines of:
- submit PR
- someone else reviews and merges (or maybe calls for someone else to have a look)

/cc @larsimmisch, @WeAreLeka, @leojrfs whaddayallthink?

PS: After having a first look I think we should use @WeAreLeka's tap as the baseline as @ladislas did a wonderful job documenting it and providing multiple versions (_if_ they are in on the idea)!

**Comment by leojrfs** [comment]
Sounds good :smile:

**Comment by ladislas** [comment]
Yep! Happy to be able to help and provide a unique and simple solution for everyone wanting to install the toolchain :)

Using my version as the baseline would be my honor :)

As for the documentation, I love writing it, so it will be a pleasure ;)

**Comment by leojrfs** [comment]
what about the name for the organization?
Do we agree on homebrew-embedded? Any other ideas?

homebrew-ucontrollers
homebrew-microcontrollers
homebrew-uc

:smile:

**Comment by jpommerening** [comment]
homebrew-embedded, homebrew-uc both sound good to me :)

**Comment by larsimmisch** [comment]
First of all, I am in favour of combining the efforts. I'll have a look at the repo of @WeAreLeka.

Regarding a change of name: I chose homebrew-avr with a reason. homebrew-embedded or homebrew-uc both sound overly ambitious to me. I don't even know how many different micro controller architectures are out there, not even the magnitude. The main competitor is Microchip with their PIC line and there still seems only sdcc to be available for it as an open source compiler, and last time I looked (which is admittedly a few years ago), sdcc was severely limited when compared to gcc. I do not see how a repo like homebrew-embedded can live up to its name.

If we do combine the repositories, I think someone should register a new git account (like homebrew-<whatever>, clone to there, and give the main contributors write access. All other repos should then update their READMEs to point to that.

**Comment by jpommerening** [comment]
If we consider embedded in general there's MIPS and ARM. If there were any usable open source solutions for PCB design on OSX I could imagine these would also fit under the "embedded" umbrella. Just my opinion though.

I think it all boils down to what you want to type before the slash in `brew tap $x/avr`c

**Comment by jpommerening** [comment]
Ohh wait, let me clarify that confusing statement (commenting via email on a phone was a bad idea). I'm not suggesting to add MIPS or ARM support or adding a formula for Kicad or anything like that. Actually, it might be to best forget the whole comment …

I think nobody was suggesting to change the name. What leo was suggesting was to name the new Github _user/org_ `homebrew-${somethinggeneric}` and that's something I can agree with, because it leaves room for "other stuff" (in other repositories).

**Comment by leojrfs** [comment]
precisely @jpommerening

I thought that since it would become homebrew-avr/homebrew-avr, and it is my intention to do the same with other uC archs, it would be better to be off with an organization that mantains uC & homebrew related stuff. So, it would eventually become organized as such:
- homebrew-(wharever the organization name we agree on) /
  - homebrew-avr
  - homebrew-pic
  - homebrew-nxpstuff
  - homebrew-freescale

And so on

There are plenty crucial tools and libs released on opensource licenses from many uC vendors, it is not only about compilers.

It is my oppinion that it should be named "-embedded" because it can relate to other stuff than microcontrollers. :smiley: 

I also think that would be good to stay with the `homebrew-` prefix in the repos names for various reasons

What is your opinion?

**Comment by ladislas** [comment]
@leojrfs it is your opinion that it **should** or **should not** be name `homebrew-embedded` ?

As it will be used mainly for cross compilation, be it on avr or pic I suggest something like:
- homebrew-cross
- homebrew-crosscompilation
- homebrew-cc 

:)

**Comment by leojrfs** [comment]
@ladislas after thinking for a bit, i rewrote this post.
Maybe `-embedded` or other generic suffix is the way to go, if we start pushing stuff related to new micros on ARM arches and stuff, it would be less than half step to push tools, cross compilers and libs for other arches and OS's, i mean not only bare metal.

anyways we should just decide, and start pushing.

What can we all agree on? please respond to these topics:
- Name for the Organization, should it even be homebrew-organizarion? or just organization (generic name related to micros or embedded and OSX)? 
  - **Re:** I think we should be better with a generic name related to OSX, cross-osx, embeddedtools-osx, whatever, this way we can have repos not only related to homebrew.
- Prefix on homebrew repos?
  - **Re:** taking on my last response, it is the way to distinguish
- Who we should tap for the baseline repo?
  - **Re:** i agree with @jpommerening on @WeAreLeka tap, since they seem to try to support multiple gcc versions. @ladislas can you extend my comment?

Please respond and post any other questions that you find important so that we can get up and running asap.

:smile:

**Comment by leojrfs** [comment]
@ladislas after thinking for a bit, i rewrote this post.
Maybe -embedded or other generic suffix is the way to go, if we start pushing stuff related to new micros on ARM arches and stuff, it would be less than half step to push tools, cross compilers and libs for other arches and OS's, i mean not only bare metal.

anyways we should just decide, and start pushing.

What can we all agree on? please respond to these topics:

Name for the Organization, should it even be homebrew-organizarion? or just organization (generic name related to micros or embedded and OSX)?

Re: I think we should be better with a generic name related to OSX, cross-osx, embeddedtools-osx, whatever, this way we can have repos not only related to homebrew.
Prefix on homebrew repos?

Re: taking on my last response, it is the way to distinguish
Who we should tap for the baseline repo?

Re: i agree with @jpommerening on @WeAreLeka tap, since they seem to try to support multiple gcc versions. @ladislas can you extend my comment?
Please respond and post any other questions that you find important so that we can get up and running asap.

:smile:

**Comment by ladislas** [comment]
My point of view:
- `osx-cross` looks nice for the the organisation name. it's simple, not restrictive and it shows that this repo is about compiling, debugging and so on, on OS X, for other platforms  
  `homebrew-embedded` could be nice as well but would mean that the organisation only maintains brew formula.
- for the homebrew repos inside the organisation, we have no choice and must use `homebrew-name_of_the_tool`
- for the baseline repo, once the organisation is created, I can change the ownership.

I indeed tried to maintain different versions of avr-gcc but it's not easy and you have to link/unlink the different versions for them to work. I have to keep working on it and see if it's possible to have something like `rbenv` for avr-gcc but it might be a lot of overhead.

**Comment by leojrfs** [comment]
:+1:

**Comment by jpommerening** [comment]
TL;DR: :+1:

> `homebrew-embedded` could be nice as well but would mean that the organisation only maintains brew formula

Is there anything besides brew formulae we want to share there?

Anyway … I'm with you guys and I don't care that much about the name. Really, pick any. We can even change the org name later if it no longer fits. So how about you go ahead, create an org+repo and _then&there_ we'll figure out what formulae to pick from which repository? For now, all we need is an org and _one_ repo. No need to get fancy.

---

PS @ladislas: (Horribly off-topic, don't reply. Think about it, we'll take care of this later.)

> I indeed tried to maintain different versions of avr-gcc but it's not easy and you have to link/unlink the different versions for them to work. I have to keep working on it and see if it's possible to have something like rbenv for avr-gcc but it might be a lot of overhead.

Did you try `configure --program-suffix=-${version}`? If multi-version support is too much of a hassle maybe we should drop it, as long as there is no explicit need for it.

**Comment by leojrfs** [comment]
https://github.com/osx-cross

lets go lets go lets go

:smile:

**Comment by marcomonteiro** [comment]
This is great.  Love the enthusiasm.
